impeller/renderer/compute_pass.h

```diff
@@ -35,6 +35,19 @@ class ComputePass : public ResourceBinder {
 
   virtual fml::Status Compute(const ISize& grid_size) = 0;
 
+  /// @brief Ensures all previously encoded commands have finished reading or
+  ///        writing to any buffer before subsequent commands will access it.
+  ///
+  ///        This only handles read after write hazards.
+  virtual void AddBufferMemoryBarrier() = 0;
+
+  /// @brief Add a barrier that ensures all previously encoded commands have
```

Comment:
Try to use the same verbage on similar API, otherwise my brain thinks "hmm this one Adds a barrier" but Buffer memory doesn't? (I'm guessing they both could be written more similar, and if not, let's clarify)

Reply:
is the `adds a barrier` part redundant? I should update this too, I was trying to describe what a barrier does instead of restating the method name.

Reply:
I think "ensures all previously encoded ..." is probably sufficient.

The fact that it does so by adding the barrier is, as you stated, is in the API name.

The only thing I'd add I suppose is when this API is supposed to be used.

Reply:
Rephrased what this does.